# HG changeset patch
# Parent  48c2f47fe0f818fcb88207efae2858d0c260631b
Feature: Linux system tray, activation for older systrays and Mate.

Code by Narthorn <narthorn@gmail.com> from
https://github.com/AyatanaIndicators/libayatana-appindicator/pull/71#issuecomment-1741638469

diff --git a/third_party/appindicator/app-indicator.c b/third_party/appindicator/app-indicator.c
--- a/third_party/appindicator/app-indicator.c
+++ b/third_party/appindicator/app-indicator.c
@@ -1983,16 +1983,17 @@ tooltip_changes (AppIndicator * self, gp
     g_free (title_escaped);
 }
 
 /* Handles the activate action by the status icon by showing
    the menu in a popup. */
 static void
 status_icon_activate (GtkStatusIcon * icon, gpointer data)
 {
+    g_signal_emit(APP_INDICATOR(data), signals[ACTIVATE_EVENT], 0, 0, 0);
     GtkMenu * menu = app_indicator_get_menu(APP_INDICATOR(data));
     if (menu == NULL)
         return;
 G_GNUC_BEGIN_IGNORE_DEPRECATIONS
     gtk_menu_popup(menu,
                    NULL, /* Parent Menu */
                    NULL, /* Parent item */
                    gtk_status_icon_position_menu,
